torch/_prims_common/__init__.py

```diff
@@ -504,23 +505,34 @@ def compute_elementwise_output_logical_to_physical_perm(
     shape = tensors[0].shape
 
     def should_swap(idx_a, idx_b):
+        def gte(a, b):
+            # semantics for a >= b, assuming a != b, a >= 0, b >= 0
+            if _guard_semantics(a == 0):
+                return False
+            if _guard_semantics(b == 0):
+                return True
+
+            expr = (a.node.expr if isinstance(a, torch.SymInt) else a) // (b.node.expr if isinstance(b, torch.SymInt) else b)
+            if isinstance(expr, (int, sympy.Integer)):
+                return expr >= 1
+            return not isinstance(expr, sympy.floor)
```

Comment:
if i understand this well 
you want to divide a//b
and check if the result is >1 or <1 
if i could not tell if the result is >1 or <1
then i want something like
u1*u2//u1 to be simplified to u1
and the inverse
u1//u1*u2 to be simplified to 1/u1
and you want a way to detect that something is >1 or <1 

1) i do not get how this floor does it :) can you explain it. 
2)  do you know that u1*u2//u1 is simplified to u2? (do we just assume u1 is not zero and simplify now?)